### ISSUE TITLE
Custom 404 code in handler causes failure at handler.lua:146

### ISSUE THREAD

**Issue by midn1** [EvandroLG/pegasus.lua#132]
As in title, doing `response:statusCode(404)` within a handler for a custom 404 response causes the `response:writeDefaultErrorMessage` call at [handler.lua:146](https://github.com/EvandroLG/pegasus.lua/blob/8e8d81d27c35ce966b22bf819faeae0bb7982eca/src/pegasus/handler.lua#L146) to crash the program.

This doesn't seem like a very niche requirement to me

**Comment by Tieske** [comment]
Yes, I have seen that as well. The problem is with the `Handler` implementation which mixes a set of different roles. To fix this however would be a breaking change.

My 2cts; the router I added in https://github.com/EvandroLG/pegasus.lua/pull/131 should not actually be a plugin, but it should replace the `Handler`. It should become the entry point for any request, and it should be possible to add plugins on a per-path basis. For example;

- `/static` -> File handler plugin, Compress plugin
- `/downloads` -> Download plugin, Compress plugin
- `/api` -> call-back based

etc.

But that would require breaking changes. Though would make for a great 1.0 version.

wdyt @EvandroLG ?

**Comment by midn1** [comment]
Though I'd prefer the ability to handle requests completely on my own, I wouldn't mind a fixed-function routing system either.

If you'd be going for the latter, route redirections will be a must (so I'd be able to, again, redirect to a 404 page)
